dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3GetJob.java

```diff
@@ -259,21 +259,26 @@ private static long getTotalJobSize(final FluentIterable<Ds3Object> obj) {
     }
 
     public FluentIterable<Ds3Object> getDS3Objects(final String bucketName, final Ds3TreeTableValueCustom selectedItem) {
-        Iterable<Contents> c;
-        try {
-            c = wrappedDs3Client.listObjects(bucketName, selectedItem.getFullName());
-        } catch (final IOException e) {
-            LOG.error("Failed to list objects", e);
-            loggingService.logMessage("Failed to list objects for " + bucketName, LogType.ERROR);
-            c = FluentIterable.from(new Contents[0]);
-        }
-        return FluentIterable.from(c).transform(contents -> {
-            if (contents != null) {
-                return new Ds3Object(contents.getKey(), contents.getSize());
-            } else {
-                return null;
+        if (selectedItem.getType() == Ds3TreeTableValue.Type.Directory) {
+            Iterable<Contents> c;
```

Comment:
This seems like it could be made final.  You may try and see it it will allow it.

Reply:
When I make it final, I become unable to assign to c in the catch case.

Reply:
Oh weird.  No worries then.